Update JaCoCo to 0.8.7

I've encountered an issue 
```
Caused by: java.lang.IllegalStateException: Unexpected SMAP line: *S KotlinDebug
```

and it is related to [this reported issue](https://github.com/jacoco/jacoco/issues/1155)  


and fortunately for me it is now fixed today in version 0.8.7! We just to updated JaCoCo!


[JaCoCo 0.8.7 has been released](https://twitter.com/_godin_/status/1389728203342757891)
[JaCoCo Changelogs](https://www.jacoco.org/jacoco/trunk/doc/changes.html)

--- a/plugin/src/test/test-fixtures/multi-module/build.gradle
+++ b/plugin/src/test/test-fixtures/multi-module/build.gradle
@@ -39,7 +39,7 @@ task clean(type: Delete) {
 
 jacoco {
     // https://www.jacoco.org/jacoco/trunk/doc/changes.html
-    toolVersion = "0.8.6"
+    toolVersion = "0.8.7"
 }
 
 rootCoverage {

